### PR TITLE
Added a new setting that doesnt show your username if you have it turned on

### PR DESCRIPTION
**Pull request by ZoamIl**

### CODE REVIEW

File: src/plugins/showMeYourName/index.tsx [hunk 1]
```diff
@@ -61,20 +67,39 @@ export default definePlugin({
...
+                    return <>{prefix}{nick}</>;
+                }
+            }
+
+            if (settings.store.mode === "user-nick") {
                 return <>{prefix}{username} <span className="vc-smyn-suffix">{nick}</span></>;
+            }
 
-            if (settings.store.mode === "nick-user")
+            if (settings.store.mode === "nick-user") {
```
Comment (Scyye) [inline src/plugins/showMeYourName/index.tsx:96]: Why change this formatting?

File: src/plugins/showMeYourName/index.tsx [hunk 2]
```diff
@@ -61,20 +67,39 @@ export default definePlugin({
...
-            if (settings.store.mode === "user-nick")
+            if (settings.store.ignoreSelf && user.id === currentUser) {
+                if (settings.store.displayNames && (user as any).globalName) {
+                    return <>{prefix}{(user as any).globalName}</>;
+                } else {
+                    return <>{prefix}{nick}</>;
+                }
+            }
+
+            if (settings.store.mode === "user-nick") {
```
Comment (Scyye) [inline src/plugins/showMeYourName/index.tsx:92]: Why change this formatting?

File: src/plugins/showMeYourName/index.tsx [hunk 3]
```diff
@@ -61,20 +67,39 @@ export default definePlugin({
...
+
+            if (settings.store.displayNames) {
                 username = (user as any).globalName || username;
+            }
 
             const { nick } = author;
             const prefix = withMentionPrefix ? "@" : "";
 
-            if (isRepliedMessage && !settings.store.inReplies || username.toLowerCase() === nick.toLowerCase())
+            if (isRepliedMessage && !settings.store.inReplies || username.toLowerCase() === nick.toLowerCase()) {
```
Comment (Scyye) [inline src/plugins/showMeYourName/index.tsx:80]: Why change this formatting?

File: src/plugins/showMeYourName/index.tsx [hunk 4]
```diff
@@ -61,20 +67,39 @@ export default definePlugin({
         try {
             const user = userOverride ?? message.author;
             let { username } = user;
-            if (settings.store.displayNames)
+
+            const currentUser = UserStore.getCurrentUser()?.id;
+
+            if (settings.store.displayNames) {
```
Comment (Scyye) [inline src/plugins/showMeYourName/index.tsx:73]: Why change this formatting?

File: src/plugins/showMeYourName/index.tsx [hunk 5]
```diff
@@ -61,20 +67,39 @@ export default definePlugin({
...
+            if (settings.store.mode === "user-nick") {
                 return <>{prefix}{username} <span className="vc-smyn-suffix">{nick}</span></>;
+            }
 
-            if (settings.store.mode === "nick-user")
+            if (settings.store.mode === "nick-user") {
                 return <>{prefix}{nick} <span className="vc-smyn-suffix">{username}</span></>;
+            }
+
+            if (settings.store.mode === "user") {
```
Comment (Scyye) [inline src/plugins/showMeYourName/index.tsx:100]: This is literally useless.

**Review by Scyye — COMMENTED** [review]
I dont exactly understand the point of this option :shrug:

File: src/plugins/showMeYourName/index.tsx (hunk 2 above)
Comment (ZoamIl) [reply src/plugins/showMeYourName/index.tsx:92]: looks better

**Comment by Vendicated** [comment]
i don't think this is very useful. also, please stick to the existing formatting and don't change it unnecessarily.

**Pull request closed without merging**